balancer-js/src/modules/data/pool/subgraphOnChain.ts

```diff
@@ -42,7 +42,6 @@ export class PoolsSubgraphOnChainRepository
 
   private filterPools(pools: Pool[]): Pool[] {
     const filteredPools = pools.filter((p) => {
-      if (p.swapEnabled === false) return false;
```

Comment:
I'm not 100% sure this is the best was to handle this. Maybe a better approach would be to create a second TestPoolHelper that does not rely on onChain data. Let me know what you think

Reply:
Right now these "ignored" pools are being filtered out and not being returned at all, but we could instead simply not update their balances and return them with whatever data the subgraph provided. Not sure this is the intended behavior though.

Reply:
The potential issue with this is it will return pools with swaps paused for other modules that use it. Right now that seems to be `PoolGraph` and I'm not sure what the effects would be.

Reply:
Got it! I just reverted this and refactored TestPoolHelper to also allow using subgraph instead of only subgraphOnChain. ✅